backend/src/main/java/com/empire/Constants.java

```diff
@@ -2,45 +2,119 @@
-    public static final double disciplinedMod = 0.1;
+    public static final String armyImpressmentTag = "Impressment";
+    public static final String armyWeatheredTag = "Weathered";
+    public static final String armyPathfindersTag = "Pathfinders";
+    public static final String armyRidersTag = "Riders";
+    public static final String armyCraftsSoldiersTag = "Crafts-soldiers";
 
-    public static final String nobleLoyalTag = "Loyal";
-    public static final double loyalMod = 0.25;
+    public static final String pirateKingdom = "Pirate";
 
     public static final double swordOfTruthMod = 0.25;
     public static final double iruhanMod = 0.15;
     public static final double lastStandMod = 4.0;
     public static final double perInspireMod = 0.05;
 
+    // Nobles
+    public static final String noblePiousTag = "Pious";
+    public static final int noblePiousFactor = 3;
+
+    public static final String nobleLoyalTag = "Loyal";
+    public static final double loyalMod = 0.25;
+    public static final double loyalMinConqMod = 1.0;
+
+    public static final String nobleInspiringTag = "Inspiring";
+    public static final double nobleInspiringMod = 0.5;
+
+    public static final String nobleUntrustingTag = "Untrusting";
+    public static final double nobleUntrustngMod = -0.35;
+
+    public static final String nobleTyrannicalTag = "Tyrannical";
+    public static final double nobleTyrannicalMod = -0.5;
+
+    public static final String nobleFrugalTag = "Frugal";
+    public static final double nobleFrugalMod = 0.5;
+
+    public static final String nobleHoardingTag = "Hoarding";
+    public static final double nobleHoardingMod = -0.35;
+
+    public static final String nobleRationingTag = "Rationing";
+    public static final double nobleRationingMod = -0.2;
+
+    public static final String nobleWastefulTag = "Wasteful";
+    public static final double nobleWastefulMod = 0.1;
+
+    public static final String noblePolicingTag = "Policing";
+
+    public static final String nobleShadyTag = "Shady Connections";
+    public static final double nobleShadyMod = 2.0;
+
+    public static final String nobleDesperateTag = "Desperate";
+    public static final double nobleDesperateMod = -2.0;
+
+    public static final double noblePirateThreatMod = -0.5;
+
+    //Religion
+    public static final double rjinkuRecruitmentMod = 1.0;
+    public static final double rjinkuBattledRecruitmentMod = 0.5;
+    public static final double syrjenTaxMod = 1.25;
+    public static final double swordOfTruthRecruitmentMod = 1.0;
+    public static final double chaliceOfCompassionTaxMod = -0.3;
+    public static final double chaliceOfCompassionFoodMod = -0.15;
+    public static final double tapestryRecruitmentMod = 0.5;
+    public static final double tapestryTaxMod = 0.5;
+    public static final double vesselOfFaithSetRelUnrestMod = -0.05; // TODO: this is used in setReligion, what rule does this implement?
```

Comment:
This is the VoF state ideology bonus, which currently reads:
"Popular unrest in all regions you control decreases by 10 percentage points whenever any region is converted to Vessel of Faith."

(So, the constant was wrong here. Should be -0.1.)